test/test_open3.rb

```diff
@@ -284,4 +284,15 @@ def test_pipeline
     }
   end
 
+  def test_pipeline_err_to_out
+    cmd = [RUBY, '-e', 'STDERR.print "err"; STDOUT.print "out"']
+    cmd_expect_only_out = [RUBY, '-e', 's = STDIN.read; exit s == "out"']
+    cmd_expect_out_and_err = [RUBY, '-e', 's = STDIN.read; exit s == "errout"']
+    result_only_out = Open3.pipeline cmd, cmd_expect_only_out
+    result_out_and_err = Open3.pipeline cmd, cmd_expect_out_and_err, {:err => :out}
+    (result_only_out + result_out_and_err).each do |status|
+      assert status.success?
+    end
```

Comment:
Although it doesn't work with Open3.pipeline when the stdout of the last command is not redirected.
`require 'open3'

cmd = ['ruby', '-e', 'STDERR.puts "err"; STDERR.flush; STDOUT.puts "out"']
opts = {
  :err => %i[child out],
}

Open3.pipeline(cmd, 'less', opts)`

open3.rb:648:in `spawn': child fd 1 is not redirected (ArgumentError)

But works well with Open3.pipeline_r when the stdout is redirected into the pipeline inside the ruby process.